### ISSUE TITLE
Create UI for adding transaction category expense code configuration

### ISSUE THREAD

**Issue by gujacq** [MidwestAccessCoalition/lockbox_rails#469]
Pretext: On the new admin dashboard being created in #349, add the UX ability to adjust this list of categories.

## Context
Give me background. 
What's happening or not happening currently?

## Process
How do I recreate what you see? Give me the steps to get here.

## Acceptance Criteria/Definition of Done
What do you want to happen?
Possible fixes?

## Resources
Add screenshots/mocks if relevant
Add relevant labels

**Comment by gujacq** [comment]
Ability to add, subtract, edit finance categories

Display associated codes for subtract and edit - users can edit associated codes as well as finance category themselves

Users input codes if they're adding a new finance category - this is a required field

Associate codes are 1:1 with categories

**Comment by gujacq** [comment]
Updated the design to include the finance categories at the bottom of the admin dashboard. I've split the admin dashboard to two tabs: one for managing users and one for managing finance categories. On the managing finance categories page, users can click on edit to edit all fields (the category name and associate code) in that row or to delete entirely: 

Manage users page (no changes except that the page is narrower and there's an additional tab to tab over to managing finance categories): https://invis.io/9EMYOUG7ZBA#/408360709_dashboard_-_MAC_Admin_-admin_Dashboard-

New manage finance categories page: https://invis.io/9EMYOUG7ZBA#/439679905_Dashboard_-_MAC_Admin_-admin_Dashboard_-_Transaction_Categories

![dashboard - MAC Admin (admin dashboard - transaction categories](https://user-images.githubusercontent.com/33872753/101289836-69053a00-37c4-11eb-98e2-54c232d1066d.png)
![dashboard - MAC Admin (admin dashboard)](https://user-images.githubusercontent.com/33872753/101289837-699dd080-37c4-11eb-8f52-963b73860e55.png)

**Comment by misternu** [comment]
@jel535s @gujacq I've been looking into how to do this, there's a lot to think about but I have a major issue that should be resolved before we move on:

If we delete or edit an existing category that has already been used, what do we expect to happen to existing support requests that use that category? In the case of an edit, should the existing support requests reflect that new category name? In the case of delete, should the existing support requests have that transaction vanish?

**Comment by jel535s** [comment]
My thoughts:
I don't think we should retroactively adjust categories in older Support Requests. So ideally, I think wherever we are storing the Support Requests would still contain those categories, but they would not be visible to the end user.
I don't think this will cause a huge problem in data bloat because we expect these changes to be few and far between (maybe one or two a year, with that decreasing as our financial reporting gets more standardized in the future).

**Comment by gujacq** [comment]
I agree with Jesse: I don't think it should retroactively adjust, and there shouldn't be anything visible to the end user. However, perhaps this does notify an admin or at the very least whoever we have attached as the accountant to alert them to a change in category... I'm thinking also whenever we have a reporting UI, some kind of notification or flag is something we can design for as well.
